src/sentry/workflow_engine/endpoints/project_detector_index.py

```diff
@@ -0,0 +1,126 @@
+from drf_spectacular.utils import PolymorphicProxySerializer, extend_schema
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NOT_FOUND,
+    RESPONSE_UNAUTHORIZED,
+)
+from sentry.apidocs.parameters import GlobalParams
+from sentry.issues import grouptype
+from sentry.workflow_engine.endpoints.serializers import DetectorSerializer
+from sentry.workflow_engine.models import Detector
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Workflows"])
+class ProjectDetectorIndexEndpoint(ProjectEndpoint):
```

Comment:
should we add a feature flag here to make sure people don't find it? (just wanted to check)

Reply:
The publish status will prevent this endpoint from showing up in the API docs on merge. A feature flag for these endpoints wouldn't hurt though.

Reply:
Discussed on slack and we won't worry about the feature for now